README.md

```diff
@@ -1,5 +1,131 @@
-# Minishift Fedora VM
+<a name="creating-a-minishift-fedora-iso"></a>
+# Minishift Fedora ISO
 
 This repository contains all the instructions and code to build a Live ISO based on Fedora
 which can be used by [minishift](https://github.com/minishift/minishift) as an alternative to
-the boot2docker and CentOS ISO.
+the boot2docker ISO.
+
+----
+
+<!-- MarkdownTOC -->
+
+- [Building the Fedora ISO](#building-the-fedora-iso)
+	- [On Fedora](#on-fedora)
+		- [Prerequisites](#prerequisites)
+		- [Building the ISO](#building-the-iso)
+	- [On hosts _other than Fedora_ \(OS X, Windows, CentOS ...\)](#non-fedora-hosts)
+		- [Prerequisites](#prerequisites-1)
+		- [Building the ISO](#building-the-iso-1)
+  - [Manual release](#manual-release)
+- [Further reading](#further-reading)
+- [Community](#community)
+
+<!-- /MarkdownTOC -->
+
+----
+
+<a name="building-the-fedora-iso"></a>
+## Building the Fedora ISO
+
+The following contains instructions on how to build the default (Fedora based) ISO.
+If you are able to install [livecd-tools](https://github.com/rhinstaller/livecd-tools)
+directly on your machine, you can use the [Fedora](#on-fedora) instructions.
+
+If you don't have _livecd-tools or using different linux distro other than fedora_, follow the
+[hosts other than Fedora](#non-fedora-hosts) instructions.
+
+<a name="on-fedora"></a>
+### On Fedora
+
+<a name="prerequisites"></a>
+#### Prerequisites
+* Update your system before start and if there is kernel update then reboot your system to activate latest kernel.
+
+        $ dnf update -y
+
+* [Install livecd-tools](https://github.com/rhinstaller/livecd-tools)
+
+  Note: We use to have docker installed on system to get selinux context, [check bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1303565)
+
+        $ dnf install -y livecd-tools docker
+
+
+<a name="building-the-iso"></a>
+#### Building the ISO
+
+```
+$ git clone https://github.com/minishift/minishift-fedora-iso.git
+$ cd minishift-fedora-iso
+$ make
+```
+
+<a name="non-fedora-hosts"></a>
+### On hosts _other than Fedora_ (OS X, Windows, CentOS ...)
+
+<a name="prerequisites-1"></a>
+#### Prerequisites
+
+* [Vagrant](https://www.vagrantup.com/)
```

Comment:
We might need to pin the Vagrant version as this is painful to see everytime user compalins this version not working etc etc. But for now it is OK IMO.